src/test/java/team4/footwithme/stadium/service/CourtServiceImplTest.java

```diff
@@ -0,0 +1,146 @@
+import team4.footwithme.stadium.repository.StadiumRepository;
+import team4.footwithme.stadium.service.response.CourtDetailResponse;
+import team4.footwithme.stadium.service.response.CourtsResponse;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@Transactional
+class CourtServiceImplTest extends IntegrationTestSupport {
+
+    @Autowired
+    private CourtRepository courtRepository;
+
+    @Autowired
+    private StadiumRepository stadiumRepository;
+
+    @Autowired
+    private MemberRepository memberRepository;
+
+    @Autowired
+    private CourtServiceImpl courtService;
+
+
+    @BeforeEach
+    void setUp() {
+        Member testMember = Member.builder()
+                .email("test@example.com")
+                .password("password")
+                .name("Test User")
+                .phoneNumber("010-1234-5678")
+                .loginType(LoginType.builder()
+                        .loginProvider(LoginProvider.ORIGINAL)
+                        .snsId("example@example.com")
+                        .build())
+                .gender(Gender.MALE)
+                .memberRole(MemberRole.USER)
+                .termsAgreed(TermsAgreed.AGREE)
+                .build();
+        memberRepository.save(testMember);
+
+        Position position1 = Position.builder()
+                .latitude(37.5642135)
+                .longitude(127.0016985)
+                .build();
+
+        Stadium stadium = Stadium.builder()
+                .name("Stadium1")
+                .address("seoul")
+                .phoneNumber("123-4567")
+                .description("Description1")
+                .position(position1)
+                .member(testMember)
+                .build();
+        stadiumRepository.save(stadium);
+
+        Court court1 = Court.builder()
+                .stadium(stadium)
+                .name("Court 1")
+                .description("Test Court 1")
+                .pricePerHour(BigDecimal.valueOf(10000))
+                .build();
+
+        Court court2 = Court.builder()
+                .stadium(stadium)
+                .name("Court 2")
+                .description("Test Court 2")
+                .pricePerHour(BigDecimal.valueOf(150000))
+                .build();
+
+        courtRepository.save(court1);
+        courtRepository.save(court2);
+    }
+
+    @Test
+    @DisplayName("유효한 stadiumId가 제공되면 Court 목록을 반환해야 한다")
+    void testGetCourtsByStadiumId_ReturnsCourts() {
+        Long stadiumId = 3L;
+
+        List<CourtsResponse> result = courtService.getCourtsByStadiumId(stadiumId);
+
+        assertThat(result).isNotNull();
+        assertThat(result).hasSize(2);
+        assertThat(result)
+                .extracting(CourtsResponse::name)
+                .containsExactly("Court 1", "Court 2");
```

Comment:
```suggestion
        assertThat(result).isNotNull()
            .hasSize(2);
            .extracting(CourtsResponse::name)
            .containsExactly("Court 1", "Court 2");
```

메서드체이닝이 가능해서 좀 더 간단하게 짤 수 있습니다 ㅎㅎ

Reply:
지난 PR 리뷰 보고 assertThat 사용하려고 한게 아직 익숙하지 않네요
감사합니다!